project_1/src/utils.cpp

```diff
@@ -0,0 +1,60 @@
+#include "utils.hpp"
+
+std::uint64_t GetCurrentTimeInMicroseconds() {
+    return std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::system_clock::now().time_since_epoch()).count();
```

Comment:
Any advantage to use the internal clock time rather than introducing an artificial (simulated) time?

Reply:
I think there are no advantages, so it will be easier and faster to use simulated time.